core/trino-main/src/main/java/io/trino/sql/planner/TypeAnalyzer.java

```diff
@@ -45,6 +56,14 @@
     private final PlannerContext plannerContext;
     private final StatementAnalyzerFactory statementAnalyzerFactory;
 
+    private final NonEvictableCache<QueryId, QueryScopedCachedTypeAnalyzer> typeAnalyzersCache = buildNonEvictableCache(
```

Comment:
It shouldn't be a cache really, but just a `Map`. We know exactly when query planning finishes, so we could discard analyzer.